### PR TITLE
Allows for distinguishing between project and item templates in GA

### PR DESCRIPTION
**Pull request by RandomlyKnighted**

### CODE REVIEW

**Comment by RandomlyKnighted** [comment]
@sayedihashimi once we merge this in we'll need to update SideWaffle to include the new TemplateType custom parameter so it gets added automatically when someone adds a new template to SideWaffle.

File: LigerShark.Templates/GoogleAnalyticsWizard.cs [hunk 1]
```diff
@@ -64,12 +66,21 @@ public bool ShouldAddProjectItem(string filePath)
             return true;
         }
 
-        private void TrackTemplate(string templateID, string templateName)
+        private void TrackTemplate(string templateID, string templateName, string templateType)
         {
             var result = GetHashString(Environment.UserDomainName + Environment.MachineName);
+            var category = "";
```
Comment (sayedihashimi) [inline LigerShark.Templates/GoogleAnalyticsWizard.cs]: I made some comments at https://github.com/RandomlyKnighted/side-waffle/commit/7fb9d25f3a60e9294c92158132bf671ca3cf9f93

**Comment by sayedihashimi** [comment]
>@sayedihashimi once we merge this in we'll need to update SideWaffle to include the new TemplateType custom parameter so it gets added automatically when someone adds a new template to SideWaffle. 

What do you mean?

**Comment by RandomlyKnighted** [comment]
For example let's say I add a new template to SideWaffle. From what I understand in our previous conversations when I add the template the wizard will get added to my vstemplate file so that we can track it as well. We'll need to update whatever adds that wizard so it has the new TemplateType custom parameter.

**Comment by sayedihashimi** [comment]
@RandomlyKnighted If I understand you right you're suggesting that we add the telemetry wizard entries into the template files for [project templates](https://github.com/ligershark/side-waffle/tree/master/TemplatePack/ItemTemplates/Extensibility/SideWaffle/SW-ProjectVSTemplateFile) and [item templates](https://github.com/ligershark/side-waffle/tree/master/TemplatePack/ItemTemplates/Extensibility/SideWaffle/SideWaffle-ItemTemplate-Basic).

Those templates are used by people adding templates to SideWaffle as well as people creating their own template packs. Because of that I don't think it's a good idea to add the wizard entries to those templates.

Instead what I think we should do is.

 1. [ ] Update the build script to check each template has telemetry values, and if missing fail the build. Since we have builds executing for PRs this will be helpful for SideWaffle
 1. [ ] Update the wiki to instruct people to add these values when contributing to SideWaffle

**Comment by RandomlyKnighted** [comment]
Perfect! If you'll update the build script I'll update the wiki article.

**Comment by sayedihashimi** [comment]
@RandomlyKnighted ok great, I have created the following issues
 - [Update build process to verify templates](https://github.com/ligershark/side-waffle/issues/327). Assigned to me.
 - [Update wiki for multi projects](https://github.com/ligershark/side-waffle/issues/328). Assigned to you.

**Comment by RandomlyKnighted** [comment]
We'll need to do this for regular templates ( not just multi project templates) as well correct?

**Pull request merged**